guides/custom_endpoints.md

```diff
@@ -0,0 +1,288 @@
+
+Because publishing is a little bit harder than subscribing and 
+there is only one endpoint that publishes track (WebRTC endpoint)
+which is also a little bit specific, we will focus on subscribing and
+create recording endpoint that will save tracks published by WebRTC endpoint.
+
+It is recommended to read [Track Lifcycle](guides/track_lifecycle.md) section before
+diving deeper in this guide.
+You will also find there some information about publishing track.
+
+## General rules
+
+Each RTC Engine endpoint has to:
+* implement `Membrane.Bin` behavior
+* has at least one Membrane element that will be responsible for producing or
+consuming data as Membrane bins are used only for logical grouping - they
+don't process data
+* publish or/and subscribe for tracks
+
+## Implementing `Membrane.Bin`
+
+To implement `Membrane.Bin` behavior we have to: 
+* `use Membrane.Bin`
+* specify some configuration options (if needed)
+* specify input and/or output pads 
+* implement some callbacks.
+
+Pads depend on what our bin is intended to do.
+For example, if our endpoint does not publish any tracks but only subscribes for tracks from other 
+endpoints it can specify only input pads.
+Pads should be in the following form
+
+```elixir
+def_input_pad :input,
+    demand_unit: :buffers,
+    caps: <caps>,
+    availability: :on_request
+
+def_output_pad :output,
+    demand_unit: :buffers,
+    caps: <caps>,
+    availability: :on_request
+```
+
+Where `caps` are `t:Membrane.Caps.t/0` or `:any`.
+
+When it comes to endpoint options we will have:
+* `rtc_engine` - pid of RTC engine. It will be used for subscribing for a track
+* `output_dir_path` - output directory path i.e. where our tracks should be saved.
+
+Putting it all together we have
+
+```elixir
+defmodule RecordingEndpoint do
+  @moduledoc """
+  Endpoint for saving tracks to a disk.
+
+  Each track will be saved in a separate file.
+  """
+  use Membrane.Bin
+
+  alias Membrane.RTC.Engine
+
+  def_options rtc_engine: [
+                spec: pid(),
+                description: "Pid of parent Engine"
+              ],
+              output_dir_path: [
+                spec: Path.t(),
+                description: "Path to directory, where tracks will be saved."
+              ]
+
+  def_input_pad :input,
+    demand_unit: :buffers,
+    caps: :any,
+    availability: :on_request
+
+  @impl true
+  def handle_init(opts) do
+    state = %{
+      directory_path: opts.directory_path,
+      rtc_engine: opts.rtc_engine
+    }
+
+    {:ok, state}
+  end
+end
+```
+
+## Consuming data
+
+Because `Membrane.Bin`s are not capable of sending or consuming any data on their own 
+(they can be more thought of as logical containers) we have to add at least one 
+casual Membrane element that will do this for us.
+
+In our case, we will need two elements:
+* [`Membrane.Stream.Serializer`](https://hexdocs.pm/membrane_stream_plugin/Membrane.Stream.Serializer.html) -
```

Comment:
The `TrackReceiver` is missing here as it's the most important item - it will be a base for the most of custom endpoints, while the elements specific to the example endpoint are, well, just an example.